src/wiring/saveEntireWiring.ts

```diff
@@ -27,56 +31,204 @@ export default async function saveEntireWiring(
+      //     },
+      //     {
+      //         "Value": "2",
+      //         "Text": "Page 2"
+      //     },
+      // ]
+      // For connector views, the "Text" is the name of the connector.
+      const pageList = await fetchPageList(
         {
           book: fetchWiringParams.book,
-          market: fetchWiringParams.contentmarket,
-          language: fetchManualParams.contentlanguage,
           cell: doc.Number,
+          title: doc.Title,
           page: "1",
-          vehicleId: fetchManualParams.vechicleId,
           bookType: fetchWiringParams.bookType,
-          country: fetchManualParams.contentmarket,
-          title: doc.Title,
+          contentmarket: fetchWiringParams.contentmarket,
+          contentlanguage: fetchManualParams.contentlanguage,
+          languageCode: fetchWiringParams.languageCode
         },
-        browserPage,
-        newTitle
+        cookieString
       );
-      continue;
+
+      // Fetch each page per Wiring section
+      for (let j = 0; j < pageList.length; j++) {
+        const page = pageList[j]["Value"];
+        const pageText = pageList[j]["Text"];
+        const sectionPath = join(wiringPath, title);
+        try {
+          await mkdir(sectionPath);
+        } catch (e: any) {
+          if (e.code !== "EEXIST") {
+            throw e;
+          }
+        }
+        console.log(
+          `Downloading wiring diagram Section: ${doc.Number} Title: ${doc.Title} Page: ${page}. `
+        );
+
+        // Query each filename, because multi-pages are not included in the TOC.
+        const pageFilename = await fetchBasicPageName(
+          {
+            environment: fetchWiringParams.environment,
+            book: fetchWiringParams.book,
+            cell: doc.Number,
+            title: doc.Title,
+            page: page,
+            bookType: fetchWiringParams.bookType,
+            contentmarket: fetchWiringParams.contentmarket,
+            contentlanguage: fetchManualParams.contentlanguage,
+          },
+          cookieString
+        );
+
+        const docStream = await fetchBasicPage(
+          pageFilename,
+          fetchWiringParams.book,
+          cookieString
+        );
+
+        await saveStream(
+          docStream,
+          join(sectionPath, `${title} - ${pageText}${doc.Filename.slice(doc.Filename.indexOf("."))}`)
+        );
+      }
     }
 
-    console.log(
-      `Downloading wiring manual page ${doc.Title} (${doc.Filename}, #${doc.Number})`
-    );
-
-    const docStream = await fetchBasicPage(
-      doc.Filename,
-      fetchWiringParams.book,
-      cookieString
-    );
-
-    await saveStream(
-      docStream,
-      join(
-        wiringPath,
-        `${sanitizedTitle} (${doc.Filename.slice(
-          0,
-          doc.Filename.indexOf(".")
-        )})${doc.Filename.slice(doc.Filename.indexOf("."))}`
-      )
-    );
+    // Start connectors
+    if (doc.Type === "Connectors" && doc.Title === "Connector Views") {
```

Comment:
Looks like we might be able to combine all these for loops into one by fetching all these lists at once then looping through every document switching on its type/title.